src/main/kotlin/no/skatteetaten/aurora/boober/feature/ApplicationDeploymentFeature.kt

```diff
@@ -11,33 +11,74 @@ import no.skatteetaten.aurora.boober.model.addEnvVarsToMainContainers
 import no.skatteetaten.aurora.boober.model.openshift.ApplicationDeployment
 import no.skatteetaten.aurora.boober.model.openshift.ApplicationDeploymentCommand
 import no.skatteetaten.aurora.boober.model.openshift.ApplicationDeploymentSpec
+import no.skatteetaten.aurora.boober.model.openshift.Notifications
+import no.skatteetaten.aurora.boober.service.AuroraDeploymentSpecValidationException
 import no.skatteetaten.aurora.boober.utils.Instants
 import no.skatteetaten.aurora.boober.utils.addIfNotNull
 import no.skatteetaten.aurora.boober.utils.durationString
 import no.skatteetaten.aurora.boober.utils.normalizeLabels
 import org.springframework.boot.convert.DurationStyle.SIMPLE
 import org.springframework.stereotype.Service
 import java.time.Duration
+import java.util.regex.Pattern
+import java.util.regex.Pattern.compile
 
 val AuroraDeploymentSpec.ttl: Duration? get() = this.getOrNull<String>("ttl")?.let { SIMPLE.parse(it) }
 
+val emailRegex: Pattern = compile(
+    "[a-zA-Z0-9\\+\\.\\_\\%\\-\\+]{1,256}" +
+        "\\@" +
+        "[a-zA-Z0-9][a-zA-Z0-9\\-]{0,64}" +
+        "(" +
+        "\\." +
+        "[a-zA-Z0-9][a-zA-Z0-9\\-]{0,25}" +
+        ")+"
+)
+
+val emailNotificationsField = "notification/email"
+val mattermostNotificationsField = "notification/mattermost"
+
 @Service
 class ApplicationDeploymentFeature : Feature {
 
     override fun handlers(header: AuroraDeploymentSpec, cmd: AuroraContextCommand): Set<AuroraConfigFieldHandler> {
         return setOf(
             AuroraConfigFieldHandler("message"),
-            AuroraConfigFieldHandler("ttl", validator = { it.durationString() })
+            AuroraConfigFieldHandler("ttl", validator = { it.durationString() }),
+            AuroraConfigFieldHandler(emailNotificationsField),
+            AuroraConfigFieldHandler(mattermostNotificationsField)
+
         )
     }
 
+    override fun validate(
```

Comment:
Burde vi egentlig ha validering at mattermost kanaler finnes? Det blir vel litt jobb egentlig.
Men kanskje vi burde ha trimming av mattermost for å luke ut whitespace feil?

Reply:
Det hørtes ikke dumt ut, kan vi løse det i oppgaven som skal sende notifications fra mattermost, for da kan det være vi har en api bruker som har tilgang tilet?

Reply:
Ja vi kan absolutt vente til da tenker jeg